exceptions/include/peach_library/exceptions/core/prefix_manager.hpp

```diff
@@ -0,0 +1,82 @@
+#ifndef PEACH_PREFIX_MANAGER_HPP
+#define PEACH_PREFIX_MANAGER_HPP
+
+#include <string>
+
+namespace peach
+{
+  enum class ExceptionTypes;
```

Comment:
Since the `::detail` part was removed in this file, this code can now be moved into the same scope as the rest